src/lib/pickles_base/proofs_verified.mli

```diff
@@ -39,23 +41,26 @@ module One_hot : sig
   val typ : (Checked.t, t) Step_impl.Typ.t
 end
 
-type 'f boolean = 'f Snarky_backendless.Cvar.t Snarky_backendless.Boolean.t
+val to_bool_vec : t -> (bool, Nat.N2.n) Vector.t
 
-(** Vector of 2 elements *)
-type 'a vec2 = ('a, Pickles_types.Nat.N2.n) Pickles_types.Vector.t
+val of_bool_vec : (bool, Nat.N2.n) Vector.t -> t
 
 module Prefix_mask : sig
-  module Checked : sig
-    type 'f t = 'f boolean vec2
-  end
-
-  val there : t -> bool vec2
+  open Kimchi_pasta_snarky_backend
 
-  val back : bool vec2 -> t
+  module Step : sig
+    module Checked : sig
+      type t = (Step_impl.Boolean.var, Nat.N2.n) Vector.t
```

Comment:
Assuming: `Step_impl.Field.Constant.t Snarky_backendless.Cvar.t Snarky_backendless.Boolean.t` is the same as `Step_impl.Boolean.var`.

Which boils down to equality of `Step_impl.Field.Constant.t` and ``Step_impl.Field.t`, if I understand correctly.
I am drowned in pickles' type pond, so I can't figure this out on my own.

And similar for `Wrap_impl`.